File: 0455. Assign Cookies.cpp
// O(n log n) time and O(1) space complexity
class Solution {
public:
    int findContentChildren(vector<int>& g, vector<int>& s) {
        sort(begin(g), end(g));
        sort(begin(s), end(s));

        int i = 0;
        int j = 0;

        while (i < g.size() && j < s.size()) {

            while (s[j] < g[i]) {
                j++;

                if (j == s.size()) {
                    return i;
                };
            };

            i++;
            j++;
        };

        return i;
    }
};

File: 0455. Assign Cookies.py
class Solution:
    # O(n log n) time and O(1) space complexity
    def findContentChildren(self, g: List[int], s: List[int]) -> int:
        g, s = sorted(g), sorted(s)
        i = j = 0

        try:
            while i < len(g):
                while s[j] < g[i]:
                    j += 1
                i += 1
                j += 1
        except IndexError:
            pass

        return i
